From 6a0f01c4ae1616844c0c2a222afa0a9c3383b07a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A5rten=20Nordheim?= <marten.nordheim@qt.io>
Date: Thu, 17 Jul 2025 13:20:53 +0200
Subject: [PATCH 416/553] HTTP: Adjust tests to handle or ignore header case

Task-number: QTBUG-137203
Pick-to: 6.8
Change-Id: I20ef2a336a8daa8be025c41f7949895d4a5143ea
Reviewed-by: Axel Spoerl <axel.spoerl@qt.io>
(cherry picked from commit c21a3afb49760016ffc822b6ee795f378ff42c66)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../access/qnetworkreply/tst_qnetworkreply.cpp   | 16 ++++++++++------
 .../access/qnetworkreply_local/minihttpserver.h  |  2 +-
 .../tst_qnetworkreply_local.cpp                  |  2 +-
 3 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/tests/auto/network/access/qnetworkreply/tst_qnetworkreply.cpp b/tests/auto/network/access/qnetworkreply/tst_qnetworkreply.cpp
index 09a880952c0..f8c2f9200ab 100644
--- a/tests/auto/network/access/qnetworkreply/tst_qnetworkreply.cpp
+++ b/tests/auto/network/access/qnetworkreply/tst_qnetworkreply.cpp
@@ -782,7 +782,7 @@ private:
 
     void parseContentLength()
     {
-        int index = receivedData.indexOf("content-length:");
+        int index = receivedData.toLower().indexOf("content-length:");
         if (index == -1)
             return;
 
@@ -3621,7 +3621,7 @@ void tst_QNetworkReply::connectToIPv6Address()
     QByteArray content = reply->readAll();
     //qDebug() << server.receivedData;
     QByteArray hostinfo = "\r\nhost: " + hostfield + ':' + QByteArray::number(server.serverPort()) + "\r\n";
-    QVERIFY(server.receivedData.contains(hostinfo));
+    QVERIFY(server.receivedData.toLower().contains(hostinfo));
     QCOMPARE(content, dataToSend);
     QCOMPARE(reply->url(), request.url());
     QCOMPARE(reply->error(), error);
@@ -8789,7 +8789,11 @@ void tst_QNetworkReply::httpUserAgent()
 
     QVERIFY(reply->isFinished());
     QCOMPARE(reply->error(), QNetworkReply::NoError);
-    QVERIFY(server.receivedData.contains("\r\nuser-agent: abcDEFghi\r\n"));
+    const char userAgentSearch[] = "\r\nuser-agent: ";
+    qsizetype userAgentIndex = server.receivedData.toLower().indexOf(userAgentSearch);
+    QCOMPARE_NE(userAgentIndex, -1);
+    userAgentIndex += sizeof(userAgentSearch) - 1;
+    QVERIFY(server.receivedData.slice(userAgentIndex).startsWith("abcDEFghi\r\n"));
 }
 
 void tst_QNetworkReply::synchronousAuthenticationCache()
@@ -8809,7 +8813,7 @@ void tst_QNetworkReply::synchronousAuthenticationCache()
                 "content-type: text/plain\r\n"
                 "\r\n"
                 "auth";
-            QRegularExpression rx("authorization: Basic ([^\r\n]*)\r\n");
+            QRegularExpression rx("[Aa]uthorization: Basic ([^\r\n]*)\r\n");
             QRegularExpressionMatch match = rx.match(receivedData);
             if (match.hasMatch()) {
                 if (QByteArray::fromBase64(match.captured(1).toLatin1()) == "login:password") {
@@ -9522,7 +9526,7 @@ void tst_QNetworkReply::ioHttpCookiesDuringRedirect()
     manager.setRedirectPolicy(oldRedirectPolicy);
 
     QVERIFY(waitForFinish(reply) == Success);
-    QVERIFY(target.receivedData.contains("\r\ncookie: hello=world\r\n"));
+    QVERIFY(target.receivedData.toLower().contains("\r\ncookie: hello=world\r\n"));
     QVERIFY(validateRedirectedResponseHeaders(reply));
 }
 
@@ -10435,7 +10439,7 @@ void tst_QNetworkReply::contentEncoding()
     {
         // Check that we included the content encoding method in our Accept-Encoding header
         const QByteArray &receivedData = server.receivedData;
-        int start = receivedData.indexOf("accept-encoding");
+        int start = receivedData.toLower().indexOf("accept-encoding");
         QVERIFY(start != -1);
         int end = receivedData.indexOf("\r\n", start);
         QVERIFY(end != -1);
diff --git a/tests/auto/network/access/qnetworkreply_local/minihttpserver.h b/tests/auto/network/access/qnetworkreply_local/minihttpserver.h
index daad88cdbcc..ae1069d7a7d 100644
--- a/tests/auto/network/access/qnetworkreply_local/minihttpserver.h
+++ b/tests/auto/network/access/qnetworkreply_local/minihttpserver.h
@@ -152,7 +152,7 @@ private:
 
     void parseContentLength(State &st, QByteArrayView header)
     {
-        qsizetype index = header.indexOf("\r\ncontent-length:");
+        qsizetype index = header.toByteArray().toLower().indexOf("\r\ncontent-length:");
         if (index == -1)
             return;
         st.foundContentLength = true;
diff --git a/tests/auto/network/access/qnetworkreply_local/tst_qnetworkreply_local.cpp b/tests/auto/network/access/qnetworkreply_local/tst_qnetworkreply_local.cpp
index fe2dc6954d9..ce2bc92e654 100644
--- a/tests/auto/network/access/qnetworkreply_local/tst_qnetworkreply_local.cpp
+++ b/tests/auto/network/access/qnetworkreply_local/tst_qnetworkreply_local.cpp
@@ -263,7 +263,7 @@ void tst_QNetworkReply_local::fullServerName()
     QVERIFY(receivedData.startsWith(expectedGet));
 
     const QByteArray expectedHost = "host: " % url.host().toUtf8() % "\r\n";
-    QVERIFY(receivedData.contains(expectedHost));
+    QVERIFY(receivedData.toLower().contains(expectedHost));
 }
 #endif
 
-- 
2.51.2

